yt_dlp/extractor/acfun.py

```diff
@@ -0,0 +1,194 @@
+import re
+from pathlib import Path
+
+from .common import InfoExtractor
+from ..utils import (
+    float_or_none,
+    int_or_none,
+    traverse_obj,
+)
+
+
+class AcFunVideoIE(InfoExtractor):
+    IE_NAME = 'AcFunVideoIE'
+    _VALID_URL = r'''(?x)
+                    https?://
+                        (?:
+                            www\.acfun\.cn/v/
+                        )
+                        ac(?P<id>[_\d]+)
+                    '''
+
+    _TESTS = [{
+        'url': 'https://www.acfun.cn/v/ac35457073',
+        'info_dict': {
+            'id': '35283657',
+            'title': '【十五周年庆】文章区UP主祝AcFun生日快乐！',
+            'duration': 455.21,
+            'timestamp': 1655289827,
+        },
+        'params': {
+            # m3u8 download
+            'skip_download': True,
+        },
+        'skip': 'Geo-restricted to China',
+    }, {
+        'url': 'https://www.acfun.cn/v/ac35457073',
+        'only_matching': True,
+    }]
+
+    @staticmethod
+    def parse_codec(codec_str):
+        vcodec = None
+        acodec = None
+
+        codec_str = codec_str or ''
+        m = re.match('(?P<vc>[^,]+),(?P<ac>[^,]+)', codec_str)
+        if m:
+            vcodec = m.group("vc")
+            acodec = m.group("ac")
+
+        return vcodec, acodec
+
+    def _real_extract(self, url):
+        mobj = self._match_valid_url(url)
+        video_id = mobj.group('id')
+
+        webpage = self._download_webpage(url, video_id)
+        json_all = self._search_json(r'window.videoInfo\s*=\s*', webpage, 'videoInfo', video_id)
+
+        video_info = json_all['currentVideoInfo']
+        playjson = self._parse_json(video_info['ksPlayJson'], video_id)
+        video_inner_id = traverse_obj(json_all, ('currentVideoInfo', 'id'))
+
+        format_jsons = traverse_obj(playjson, ('adaptationSet', 0, 'representation'))
+
+        try:
+            ext = Path(video_info['fileName']).suffix[1:]
+        except Exception:
+            self.report_warning('Parse ext failed, use mp4')
+            ext = 'mp4'
```

Comment:
I think this can handled by `determine_ext` in `utils` but i didn't test, please correct me if i'm wrong

Reply:
ok, it works